sdk/core/azure-core/azure/core/pipeline/policies/_redirect.py

```diff
@@ -55,13 +55,18 @@ def __init__(self, **kwargs):
 
     @classmethod
     def no_redirects(cls):
-        """Disable redirects."""
+        """Disable redirects.
+
+        :return: A redirect policy with redirects disabled.
+        :rtype: ~azure.core.pipeline.policies.RedirectPolicy or ~azure.core.pipeline.policies.AsyncRedirectPolicy
```

Comment:
We should type `cls` correctly, e.g.: 

```python
T = typing.TypeVar('T')

@classmethod
def no_redirects(cls: typing.Type[T]) -> T:
    return cls(redirect=False)
```

Reply:
It's not a PR about Typing, but agree on the principle

Reply:
My (implicit) suggestion is that the `rtype` should be inferred from the type hints/is the generic type parameter. I don't think there is a ReSTructured text way of representing it.

Reply:
I don't think so, it's why I decided to be explicit here. There are a few places like that like the fact that we have 2 HttpRequest impl (for this one I decided to document the "rest" one, since it's the one most likely people will see nowadays).